src/libs/actions/OpenAI/index.ts

```diff
@@ -1,6 +1,287 @@
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-function initializeOpenAIRealtime(adminsReportID: number) {}
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import type {SendRecapInAdminsRoomParams} from '@libs/API/parameters';
+import {SIDE_EFFECT_REQUEST_COMMANDS, WRITE_COMMANDS} from '@libs/API/types';
+import {playStreamSound} from '@libs/Sound';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {TalkToAISales} from '@src/types/onyx';
+import type {OnyxData} from '@src/types/onyx/Request';
 
-function stopConnection() {}
+type ConnectionResult = {
+    peerConnection: RTCPeerConnection;
+    dataChannel: RTCDataChannel;
+};
+
+type WebRTCConnections = {
+    openai: ConnectionResult | null;
+};
+
+type OpenAIRealtimeMessage = {
+    name: string;
+    arguments: string;
+    type: string;
+};
+
+type Recap = {
+    recap: string;
+};
+
+let currentAdminsReportID: number | null = null;
+let mediaStream: MediaStream | null = null;
+let clientSecret: TalkToAISales['clientSecret'];
+
+const connections: WebRTCConnections = {
+    openai: null,
+};
+
+Onyx.connect({
+    key: ONYXKEYS.TALK_TO_AI_SALES,
+    callback: (value) => {
+        if (!value) {
+            return;
+        }
+
+        clientSecret = value.clientSecret;
+    },
+});
+
+function isExpiredToken(expiresAt: number): boolean {
+    const currentUTCEpochTime = Math.floor(Date.now() / 1000);
+    return currentUTCEpochTime >= expiresAt;
+}
+
+function getEmphemeralToken(): Promise<string> {
+    if (clientSecret && !isExpiredToken(clientSecret.expiresAt)) {
+        Onyx.merge(ONYXKEYS.TALK_TO_AI_SALES, {
+            isLoading: true,
+        });
+        return Promise.resolve(clientSecret.ephemeralToken);
+    }
+
+    const onyxData: OnyxData = {
+        optimisticData: [
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.TALK_TO_AI_SALES}`,
+                value: {
+                    isLoading: true,
+                },
+            },
+        ],
+    };
+
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    return API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.GET_EMPHEMERAL_TOKEN, {}, onyxData)
```

Comment:
we still need to use `makeRequestWithSideEffects` because we need to wait for the api call to complete. and then access the token from api response